libminifi/include/core/state/Value.h

```diff
@@ -188,6 +264,18 @@ class BoolValue : public Value {
     }
   }
 
+  virtual bool getValue(uint32_t &ref) {
```

Comment:
Please, no!

Yes, it was already horrible, but that doesn't justify copy-pasting the same! :)

```
template<typename Integral, typename std::enable_if<
    (sizeof(Integral) >= 4) && std::is_integral<Integral>::value
    ,Integral>::type* = nullptr>
virtual bool getValue(Integral &ref) {
  if(ref != 0 && ref != 1) {
   return false;
  }
 ref = (ref == 1);
 return true;
}
```

Reply:
The implementation is also bad smell, I agree, didn't check that. 

In case the functions are virtual we should introduce a templated private helper and call that from the virtuals. Still better than 4 times duplicated code.